system-addon/lib/PrefsFeed.jsm

```diff
@@ -32,11 +33,13 @@ this.PrefsFeed = class PrefsFeed {
     this._storage = new ActivityStreamStorage("sectionPrefs");
   }
 
-  // If the any prefs are set to something other than what the prerendered version
-  // of AS expects, we can't use it.
-  async _setPrerenderPref() {
+  // If any prefs or the theme are set to something other than what the
+  // prerendered version of AS expects, we can't use it.
+  async _setPrerenderPref(theme) {
     const indexedDBPrefs = await this._storage.getAll();
-    this._prefs.set("prerender", PrerenderData.arePrefsValid(pref => this._prefs.get(pref), indexedDBPrefs));
+    const prefsAreValid = PrerenderData.arePrefsValid(pref => this._prefs.get(pref), indexedDBPrefs);
+    const themeIsDefault = (theme || this.store.getState().Theme).className === INITIAL_STATE.Theme.className;
+    this._prefs.set("prerender", prefsAreValid && themeIsDefault);
```

Comment:
This means dark theme is never prerendered, so potentially the page will flash all white before switching to dark?

@k88hudson any suggestions here? I suppose we could maybe have `aboutNewTabService` point to a default vs dark theme prerendered version? Although that seems a bit excessive to set a single class name or not.

@rlr also, how are theme changes going to be handled going forwards (for the later in 61 vs later)?

Reply:
> also, how are theme changes going to be handled going forwards (for the later in 61 vs later)?

You mean the design process or something else?

> This means dark theme is never prerendered, so potentially the page will flash all white before switching to dark?

There are so many things that invalidate prerender now that I'm not sure it is even useful other than first run.

Reply:
I was referring to the not needing ThemeFeed. Sounds like going forwards, we won't need to invalidate prerendering with dark theme as each content page will be getting an async message anyway to set the theme.

Hopefully preloading will avoid most of the potential flashing of wrong-theme-styling.

So with that, maybe we don't need to complicate prerendering related stuff with this PR?

Reply:
Invalidation is probably the right call, yep. 

I'm definitely noticing some flashing, and even prerendering aside if the message doesn't come through immediately you might see some flashing of a white background

I guess you could have a *single* variant (like an unprerendered, dark background version of the html) for dark theme... but I agree, preloading will take care of a lot of the problem.

Maybe we should check the delay on a slower/reference machine to see if it's acceptable

Reply:
Maybe an alternative to invalidating prerendering or extra prerendering dark theme.. What if we used `localStorage` or something in the page to add `.dark-theme` to `<body>` as the page loads? The storage gets updated when `Theme.className` arrives so the next load of the page should synchronously render the appropriate background color. Although this would prevent different themes per window…

I guess we'll just keep the invalidation for now…